### PR TITLE
Make new_style the default error capturing mode

### PR DESCRIPTION
**Pull request by sklam**
as per schedule in https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-old-style-numba-captured-errors

### CODE REVIEW

File: numba/core/config.py [hunk 1]
```diff
@@ -424,7 +424,7 @@ def avx_default():
 
         CAPTURED_ERRORS = _readenv("NUMBA_CAPTURED_ERRORS",
                                    _validate_captured_errors_style,
-                                   'old_style')
+                                   'default')
```
Comment (gmarkall) [inline numba/core/config.py:427]: This confuses the functions `use_new_style_errors()` and `use_old_style_errors()`, which only recognise `"new_style"` and `"old_style"` respectively, so neither returns `True` now.

**Review by gmarkall — APPROVED** [review]
The parts I didn't write have my approval.

**Comment by sklam** [comment]
@gmarkall changes looks good. But, I can't approve my own PR so someone else has to do it.

**Review by esc — APPROVED** [review]

**Comment by esc** [comment]
> @gmarkall changes looks good. But, I can't approve my own PR so someone else has to do it.

done

**Pull request merged**